### DISCUSSION TITLE
VPP container and minimal clab lab

### DISCUSSION THREAD

**Discussion by DanPartelly** [ipspace/netlab#1363] [General]
Turns out is pretty trivial do drive veth pairs with vpp. They  have a interface type which drives Linux host interfaces with AF_PACKET. Also, a nice man in containerlab discord pointed me to the fact that Ligato maintains a fd.io docker container. So even the minimal pain to write a Dockerfile  is gone. 

So, here is a minimal proof of concept lab (vpp only, no linux control plane, no FRR): 

https://github.com/DanPartelly/clab-minimal-vpp

Once the lab is deployed:
docker exec clab-vpp01-r1 vppctl show int addr
docker exec clab-vpp01-r1 vppctl ping 10.1.0.2

Ivan, what do you think next steps should be to bring it to networklab ? Work first time on linux CP integration and FRR in containerlab, and then proceed with networklab things ?

**Comment by ipspace** [comment]
The proof-of-concept setup is in the **vpp** branch. However, the VPP container crashed my server (have no idea why that happens, Cumulus Linux container sometimes does the same), and as containers are restarted after the reboot (now I know why that has to be fixed 🤦‍♂️) it almost bricked my NUC.

Hope that branch will help you get started. I'm not touching that container in the near future -- I would have to build a sandbox VM and I have a few other things to fix first.

**Reply by DanPartelly** [reply]
Understandable. Focus on whats important. Ill take that branch up after IOL. Thanks a lot for all your efforts. 

I will also build a vpp VM for Vagrant when Ill start working in networklab on that branch. I have weirdness with Cummuls containers, they basically take over my machine like The Borg.  I get thrown into an ever restarting debian login console.  CtrlAltDel .... Weird.
